building/configuring_git_workflow.adoc

```diff
@@ -0,0 +1,426 @@
+
+[source,bash]
+----
+[diff]
+  editor = /opt/sublimetext2/sublime_text
+  tool = bc3
+
+[merge]
+  editor = /opt/sublimetext2/sublime_text
+  tool = bc3
+----
+
+=== Useful bash scripts
+
+The following bash function will help you quickly identify when you are inside a folder
+that is recognized as a git repository.  It will customize your bash prompt to include 
+the repository name and the current branch. 
+
+[source,bash] 
+----
+# returns the branch name
+__git_ps1 ()
+{
+  local b="$(git symbolic-ref HEAD 2>/dev/null)";
+  if [ -n "$b" ]; then
+    printf " (%s)" "${b##refs/heads/}";
+  fi
+}
+export PS1='[\u@\h \W] $(__git_ps1 "(%s)")$'
+
+# this may also be required for your distro
+PROMPT_COMMAND=__git_ps1
+----
+
+Now when in a git folder, your bash prompt will look as follows:
+
+[source,bash] 
+----
+[user@localhost jbosstools-server]  (master) $
+----
+
+== Sources
+
+=== How to clone a repo to fetch sources
+
+To fetch sources, you must clone the repo to your local disk.
+
+[source,bash]
+----
+cd ~/my-github-stuff
+git clone git@github.com:jbosstools/jbosstools-base.git
+cd jbosstools-base
+----
+
+See also http://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository#Cloning-an-Existing-Repository[Cloning an Existing Repository]
+
+Then, if you plan to contribute a pull request, you will need your own fork. To fork the repo, 
+browse to the repo on github, and click the `Fork` button in the top-right corner.
+
+image::../images/fork-a-repo.png[Fork a github repo]
+
+=== How to import projects into Eclipse
+
+Launch Eclipse, then select `File -> Import -> Maven -> Existing Maven Projects`
+
+image::../images/import-maven-project.png[]
+
+Browse for where you cloned the repo with which you want to work.
+
+image::../images/import-maven-project-2.png[]
+
+Some projects require a full Maven build before they will compile properly in Eclipse with PDE. To solve this problem, you can either build using Maven via the commandline, or configure Eclipse to use m2e and Tycho to build the project instead of PDE. The import wizard will do this for you automatically.
+
+image::../images/import-maven-project-3-tycho.png[]
+image::../images/import-maven-project-4-tycho.png[]
+
+Note that the Tycho configurator also includes PDE, so that it can be installed even if you're using a version of Eclipse which doesn't already include it.
+
+image::../images/import-maven-project-5-tycho-pde.png[]
+
+=== How to add forks to your local clone
+
+If you use https://hub.github.com/[hub], you can simply add a fork like this:
+
+[source,bash]
+----
+cd ~/my-github-stuff/jbosstools-base
+git remote add -p ${gituser}
+----
+
+Or, without https://hub.github.com/[hub], do this:
+
+[source,bash]
+----
+cd ~/my-github-stuff/jbosstools-base
+git remote add ${gituser} git@github.com:${gituser}/jbosstools-base.git
```

Comment:
I just did above, but let me try again with bullets:

a) you can copy/paste the http url straight from the browser url (no need to fiddle with the git/ssh UI on github.com)
b) git hub uses https as default when using fork etc. so it fits nicely with that 
c) it does not require ssh agent running to remember passwords and git provides credential.helper for more portable setup (http://stackoverflow.com/questions/5343068/is-there-a-way-to-skip-password-typing-when-using-https-github)
d) it is the recommended way from github (https://help.github.com/articles/which-remote-url-should-i-use/) with the following reasons:
  1) https works *everywhere* even behind proxy
  2) https is available on all repos  (public or private) 
  3) https automatically knows if you have read, read/write based on your permissions (same setup for committers as well as contributors)
 
Hope that explains it.

And based on your comments above also seem to remove need for using -p/prune.